zephyr/drivers/can/can_sccan.c

```diff
@@ -404,6 +415,74 @@ static void sc_can_rx_isr(const struct device *dev)
 	return;
 }
 
+static void sc_can_get_error_count(const struct sc_can_cfg *config,
+			      struct can_bus_err_cnt *err_cnt)
+{
+	uint32_t errcnt_reg;
+
+	errcnt_reg = sc_can_read32(config, SCCAN_ECNTR_OFFSET);
+	err_cnt->tx_err_cnt = SCCAN_TXECNT(errcnt_reg);
+	err_cnt->rx_err_cnt = SCCAN_RXECNT(errcnt_reg);
+}
+
+static int sc_can_get_state(const struct device *dev, enum can_state *state,
+			      struct can_bus_err_cnt *err_cnt)
+{
+	const struct sc_can_cfg *config = dev->config;
+	uint32_t status_reg;
+
+	status_reg = sc_can_read32(config, SCCAN_STSR_OFFSET);
+	switch (SCCAN_ESTS(status_reg)) {
+		case SCCAN_ESTS_CAN_DISABLE:
+			*state = CAN_STATE_STOPPED;
+			break;
+		case SCCAN_ESTS_ERROR_ACTIVE:
+			if (status_reg & SCCAN_EWRN) {
+				*state = CAN_STATE_ERROR_WARNING;
+			} else {
+				*state = CAN_STATE_ERROR_ACTIVE;
+			}
+			break;
+		case SCCAN_ESTS_ERROR_PASSIVE:
+			*state = CAN_STATE_ERROR_PASSIVE;
+			break;
+		case SCCAN_ESTS_BUS_OFF:
+		default:
+			*state = CAN_STATE_BUS_OFF;
+			break;
+	}
+
+	sc_can_get_error_count(config, err_cnt);
```

Comment:
What is the point of calling this here?

Reply:
`sc_can_get_state` has the out parameter named as  `err_cnt`, so need to set the error count to output parameter.